tests/MongoDB.Driver.Tests/MongoCollectionProseTests.cs

```diff
@@ -0,0 +1,83 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Threading;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver.Core.Bindings;
+using MongoDB.Driver.Core.Clusters.ServerSelectors;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.TestHelpers;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using Xunit;
+
+namespace MongoDB.Driver.Tests
+{
+    public class MongoCollectionProseTests
+    {
+        [SkippableFact]
+        public void WriteConcernError_errInfo_should_be_propagated()
+        {
+            RequireServer.Check().Supports(Feature.FailPointsFailCommand);
+
+            var failPointCommand = @"
+                {
+                    ""configureFailPoint"" : ""failCommand"",
+                    ""data"" : {
+                        ""failCommands"" : [""insert""],
+                        ""writeConcernError"" : {
+                            ""code"" : 100,
+                            ""codeName"" : ""UnsatisfiableWriteConcern"",
+                            ""errmsg"" : ""Not enough data-bearing nodes"",
+                            ""errInfo"" : {
+                                ""writeConcern"" : {
+                                    ""w"" : 2,
+                                    ""wtimeout"" : 0,
+                                    ""provenance"" : ""clientSupplied""
+                                }
+                            }
+                        }
+                    },
+                    ""mode"": { ""times"": 1 }
+                }";
```

Comment:
What you have written is 100% fine but just wanted to point out that it can be done with less quoting which can be easier to read and write:

    var failPointCommand = @"
        {
            configureFailPoint : 'failCommand',
            data : {
                failCommands : ['insert'],
                writeConcernError : {
                    code : 100,
                    codeName : 'UnsatisfiableWriteConcern',
                    errmsg : 'Not enough data-bearing nodes',
                    errInfo : {
                        writeConcern : {
                            w : 2,
                            wtimeout : 0,
                            provenance : 'clientSupplied'
                        }
                    }
                }
            },
            mode: { times: 1 }
        }";

Reply:
Even less quoting is required if you want:

    var failPointCommand = @"
        {
            configureFailPoint : 'failCommand',
            data : {
                failCommands : ['insert'],
                writeConcernError : {
                    code : 100,
                    codeName : 'UnsatisfiableWriteConcern',
                    errmsg : 'Not enough data-bearing nodes',
                    errInfo : {
                        writeConcern : {
                            w : 2,
                            wtimeout : 0,
                            provenance : 'clientSupplied'
                        }
                    }
                }
            },
            mode : { times: 1 }
        }";

Reply:
I would prefer adding quotes regardless it's possible or not to avoid them for a node name/value. it's simpler :)

Reply:
I would say it's simpler to omit unnecessary quotes for field names. It's also common practice.

For example, here's a snippet from the MongoDB Shell  documentation:

    db.collection.find( { qty: { $gt: 4 } } )

Reply:
in this case, we need to remember the json rules for this case since sometimes we can avoid quotes but sometimes no. Maybe it's not a bad thing to have, but it's a definitely is not a target of this code :)

Reply:
Why is not relevant to this pull request?

You are adding new code and you can/should use our usual conventions for new code.

Reply:
As far as I know, both formats are recommended and there is no recommendation to avoid any of them (at least I didn't find any recommendation about it after quick search, can you find some guides for it?).
I'm not sure that we strongly use only this approach. But If so, let's add this to this doc: https://docs.google.com/document/d/1cBOz2Zr39yDV9FBwBVKGG9SX8P-zKBcRsLvAmp_-gF8.
My point that since there are no recommendations to avoid quotes, maybe it's more correct just adding them in any case, it will save 2-3 Kbytes of our brain memory when we will think about it each time :)

Reply:
I can tell you with absolute certainty that from the very beginning of the development of the C# driver it has been our convention to not quote field names unnecessarily in JSON string constants. But like all our conventions not always followed... :(

I can also tell you that almost all MongoDB documentation adopts this convenient convention as well.

I wouldn't expect to find any guides about this on the web. This is a convention that most likely is specific to MongoDB. In fact, not quoting field names is a MongoDB extension to JSON (that may also be common elsewhere but I don't know).

It's not about saving bytes... it's about making the strings more readable (to humans) by removing the clutter of unnecessary quotes.

Reply:
done